drizzlepac/generate_final_product_filenames.py

```diff
@@ -124,29 +279,32 @@ def total_detection_product_filename_generator(obs_info,nn):
     obs_info : list
         list of items that will be used to generate the filenames: proposal_id,
         visit_id, instrument, and detector
-    nn : string
+    tdp : string
         the single-exposure image number (NOTE: only used in
         single_exposure_product_filename_generator())
 
     Returns
     --------
     product_filename_dict : dictionary
         A dictionary containing the generated filenames.
-    """
+
+    Clarification of variables:
     proposal_id = obs_info[0]
-    visit_id = obs_info[1]
-    instrument = obs_info[2]
-    detector = obs_info[3]
+    visit_id    = obs_info[1]
+    instrument  = obs_info[2]
+    detector    = obs_info[3]
+    """
 
     product_filename_dict = {}
-    product_filename_dict["image"] = "hst_{}_{}_{}_{}.fits".format(proposal_id, visit_id, instrument, detector)
-    product_filename_dict["source catalog"] = product_filename_dict["image"].replace(".fits",".cat")
+    product_filename_dict["image"] = "hst_{}_{}_{}_{}.fits".format(obs_info[0], obs_info[1],
+                                                                   obs_info[2], obs_info[3])
+    product_filename_dict["source catalog"] = product_filename_dict["image"].replace(".fits", ".cat")
```

Comment:
An alternative:
```python
basename = 'hst_' + '_'.join(map(str, obs_info))
product_filename_dict["image"] = basename + '.fits'
product_filename_dict["source catalog"] = basename + '.cat'
```

Reply:
Done for this instance and all similar instances.